### DISCUSSION TITLE
ZFS disk not showing

### DISCUSSION THREAD

**Discussion by mcondren** [aristocratos/btop#248] [Q&A] [answered]
Hi All,
I love the utility, but I have one item I am trying to figure out.  For those of you who have a ZFS zpool on your NAS, they don't normally add an entry to fstab.  For example I have a 2TB drive that is formatted with ZFS and has a zpool called "backup".  It is a completely separate disk from my system disk. There is no /dev/disk entry present for that drive at all when you run a df.  It just automounts the zpool as /backup as the mountpoint.  I am assuming due to this, btop can't detect it as a drive.  Any ideas how to get stats on my zpool or even ZFS disk as a whole?

```
zpool list
NAME     SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
backup  1.81T   640G  1.19T        -         -     6%    34%  1.00x    ONLINE  -
```

```
Filesystem      Size  Used Avail Use% Mounted on
udev            7.8G     0  7.8G   0% /dev
tmpfs           1.6G  3.7M  1.6G   1% /run
/dev/nvme0n1p2  442G   27G  393G   7% /
tmpfs           7.8G  4.0K  7.8G   1% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
ramdisk         1.0G     0  1.0G   0% /tmp/cache
/dev/nvme0n1p1  511M  3.5M  508M   1% /boot/efi
tmpfs           1.6G     0  1.6G   0% /run/user/0
backup          1.2T  224K  1.2T   1% /backup
backup/docker   1.2T  1.3G  1.2T   1% /backup/docker
backup/frigate  1.8T  630G  1.2T  36% /backup/frigate
backup/files    1.2T  8.8G  1.2T   1% /backup/files
```

**Comment by hugolopolus** [comment]
Hi,

post back a `zpool status -v backup` - thanks

Hugo

**Comment by mcondren** [comment]
Sure @hugolopolus Please see below. 
```
❯ zpool status -v backup
  pool: backup
 state: ONLINE
  scan: scrub repaired 0B in 02:40:19 with 0 errors on Sun Jan 23 09:27:20 2022
config:

        NAME        STATE     READ WRITE CKSUM
        backup      ONLINE       0     0     0
          sda       ONLINE       0     0     0

errors: No known data errors
```

**Comment by hugolopolus** [comment]
Ok so it's mounted on /dev/sda (seemingly a local SSD/SATA drive)

I found this 'bug' posting for the project about [ZFS](https://github.com/aristocratos/btop/issues/201), so for the moment I doubt you'll get alot of useful readings out of it , but, to make it at least appear you could try: 

Menu > Options > Mem > Show Only Physical (set to False)  ## note: yes Disk settings under Memory (both storage)

Additional evidence of 'zfs' issues https://github.com/aristocratos/btop/discussions/156 - seems there isn't currently a nice solution for zfs volumes but it's not being ignored :vulcan_salute:

**Comment by mcondren** [comment]
Thanks, I’ll give that a try.

**Comment by mcondren** [comment, marked as answer]
OK, I wanted to respond back here to post how I got this working in case anyone else has the same issue.  @hugolopolus it seemed to be a combination of the Physical only, and the fstab setting.  Rather than change Physical only to "false", I had to leave it as "True" but also set read from fstab as "False".  Only that True and False combination works to show the zfs pools.    If I change to True/True or False/False it doesn't work.
